lib/quicly.c

```diff
@@ -2184,39 +2204,76 @@ static int on_ack_ack(quicly_conn_t *conn, const quicly_sent_packet_t *packet, q
     return 0;
 }
 
-static int on_ack_stream(quicly_conn_t *conn, const quicly_sent_packet_t *packet, quicly_sent_t *sent, quicly_sentmap_event_t event)
+static int on_ack_stream_acked(quicly_conn_t *conn, quicly_stream_id_t stream_id, quicly_sendstate_sent_t *sent, int is_active)
 {
     quicly_stream_t *stream;
     int ret;
 
-    if (event == QUICLY_SENTMAP_EVENT_EXPIRED)
+    if ((stream = quicly_get_stream(conn, stream_id)) == NULL)
         return 0;
 
-    if (event == QUICLY_SENTMAP_EVENT_ACKED) {
-        QUICLY_PROBE(STREAM_ACKED, conn, probe_now(), sent->data.stream.stream_id, sent->data.stream.args.start,
-                     sent->data.stream.args.end - sent->data.stream.args.start);
-    } else {
-        QUICLY_PROBE(STREAM_LOST, conn, probe_now(), sent->data.stream.stream_id, sent->data.stream.args.start,
-                     sent->data.stream.args.end - sent->data.stream.args.start);
+    size_t bytes_to_shift;
+    if ((ret = quicly_sendstate_acked(&stream->sendstate, sent, is_active, &bytes_to_shift)) != 0)
+        return ret;
+    if (bytes_to_shift != 0)
+        stream->callbacks->on_send_shift(stream, bytes_to_shift);
+    if (stream_is_destroyable(stream)) {
+        destroy_stream(stream, 0);
+    } else if (stream->_send_aux.reset_stream.sender_state == QUICLY_SENDER_STATE_NONE) {
+        resched_stream_data(stream);
     }
 
-    /* TODO cache pointer to stream (using a generation counter?) */
-    if ((stream = quicly_get_stream(conn, sent->data.stream.stream_id)) == NULL)
+    return 0;
+}
+
+static int on_ack_stream_sumbit_delayed(quicly_conn_t *conn)
```

Comment:
```suggestion
static int on_ack_stream_submit_delayed(quicly_conn_t *conn)
```

Reply:
In 0922549, renamed the functions as discussed.